core/src/code.rs

```diff
@@ -185,6 +187,8 @@ impl Code {
         module: Option<Module>,
         instrument_with_const_rules: bool,
     ) -> Result<Self, CodeError> {
+        wasmparser::validate(&original_code).map_err(|_| CodeError::Decode)?;
```

Comment:
As I can see in the comment:
```rust
    /// Create the code without checks.
```

Reply:
Agree however as you can see it contains several checks anyway. So I suppose that original code should be valid wasm-binary.